tariff/ostrom.go

```diff
@@ -185,29 +186,28 @@ func (t *Ostrom) GetContracts() ([]ostrom.Contract, error) {
 // Unfortunately, the API does not allow to query the price for these yet.
 func (t *Ostrom) runStatic(done chan error) {
 	var once sync.Once
-	var err error
-	var marketprice float64
 
 	tick := time.NewTicker(time.Hour)
 	for ; true; <-tick.C {
-		marketprice, err = t.getFixedPrice()
-		if err == nil {
-			ts := now.BeginningOfDay().Local()
-			data := make(api.Rates, 48)
-			for i := range data {
-				data[i] = api.Rate{
-					Start: ts,
-					End:   ts.Add(time.Hour),
-					Price: marketprice / 100.0,
-				}
-				ts = data[i].End
-			}
-			mergeRates(t.data, data)
-			once.Do(func() { close(done) })
-		} else {
+		price, err := t.getFixedPrice()
+		if err != nil {
 			once.Do(func() { done <- err })
 			t.log.ERROR.Println(err)
+			continue
+		}
+
+		data := make(api.Rates, 48)
+		for i := range data {
+			ts := now.BeginningOfDay().Add(time.Duration(i) * time.Hour)
```

Comment:
Ich verstehe ja, dass unsere Auffassungen von gutem Code sich unterscheiden.
Aber warum ist es besser, hier 96 Funktionsaufrufe und 48 Multiplikationen zu machen, nur um Werte zu berechnen, die du in der jeweils vorhergehenden Iteration bereits berechnet hast?
Es gilt doch data[i].Start = data[i-1].End

Reply:
Hilft uns denn die Diskussion? Am Ende muss das Coreteam den Code warten und Geschmack ist subjektiv.

Reply:
Im Prinzip ja, aber so eine Multiplikation (oder 48) alle 2 Stunden interessiert schon lange keine CPU mehr. Dann lieber lesbar, Stichwort „premature optimization“. Das ist aber sicher kein Hügel, auf dem ich sterben möchte 😅